tower-util/src/lib.rs

```diff
@@ -76,6 +78,161 @@ pub trait ServiceExt<Request>: tower_service::Service<Request> {
+    /// #       DatabaseService  
+    /// #   }
+    /// # }
+    /// #
+    /// struct Record {
+    ///    pub name: String,
+    ///    pub age: u16
+    /// }
+    ///
+    /// # impl Service<u32> for DatabaseService {
+    /// #   type Response = Record;
+    /// #   type Error = u8;
+    /// #   type Future = futures_util::future::Ready<Result<Record, u8>>;
+    /// #
+    /// #   fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+    /// #       Poll::Ready(Ok(()))
+    /// #   }
+    /// #
+    /// #   fn call(&mut self, request: u32) -> Self::Future {
+    /// #       futures_util::future::ready(Ok(Record { name: "Jack".into(), age: 32 }))
+    /// #   }
+    /// # }
+    /// #
+    /// fn main() {
+    ///     // A service returning Result<Record, _>
+    ///     let service = DatabaseService::new("127.0.0.1:8080");
+    ///
+    ///     // Map the response into a new response
+    ///     let mut new_service = service.map_ok(|record| record.name);
+    ///
+    ///     async {
+    ///         let id = 13;
+    ///         let name = new_service.call(id).await.unwrap();
+    ///     };
+    /// }
+    /// ```
+    fn map_ok<F, Response>(self, f: F) -> MapOk<Self, F>
+    where
+        Self: Sized,
+        F: FnOnce(Self::Response) -> Response + Clone,
+    {
+        MapOk::new(self, f)
+    }
+
+    /// # Example
+    /// ```
+    /// # use std::task::{Poll, Context};
+    /// # use tower_service::Service;
+    /// use tower_util::ServiceExt;
+    ///
+    /// # struct DatabaseService;
+    /// # impl DatabaseService {
+    /// #   fn new(address: &str) -> Self {
+    /// #       DatabaseService  
+    /// #   }
+    /// # }
+    /// #
+    /// struct Error {
+    ///    pub code: u32,
+    ///    pub message: String
+    /// }
+    ///
+    /// # impl Service<u32> for DatabaseService {
+    /// #   type Response = String;
+    /// #   type Error = Error;
+    /// #   type Future = futures_util::future::Ready<Result<String, Error>>;
+    /// #
+    /// #   fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
+    /// #       Poll::Ready(Ok(()))
+    /// #   }
+    /// #
+    /// #   fn call(&mut self, request: u32) -> Self::Future {
+    /// #       futures_util::future::ready(Ok(String::new()))
+    /// #   }
+    /// # }
+    /// #
+    /// fn main() {
+    ///     // A service returning Result<_, Error>
+    ///     let service = DatabaseService::new("127.0.0.1:8080");
+    ///
+    ///     // Map the error into a new error
+    ///     let mut new_service = service.map_err(|err| err.code);
+    ///
+    ///     async {
+    ///         let id = 13;
+    ///         let code = new_service.call(id).await.unwrap_err();
+    ///     };
+    /// }
+    /// ```
+    fn map_err<F, Error>(self, f: F) -> MapErr<Self, F>
+    where
+        Self: Sized,
+        F: FnOnce(Self::Error) -> Error + Clone,
+    {
+        MapErr::new(self, f)
+    }
+
+    /// # Example
```

Comment:
And again, a one-two sentence summary would be good.

Reply:
For both of these I've almost copy + pasted the summaries from the futures crate.